Source/Core/Core/ConfigLoaders/NetPlayConfigLoader.cpp

```diff
@@ -146,6 +148,11 @@ class NetPlayConfigLayerLoader final : public Config::ConfigLayerLoader
     }
 #endif
 
+#ifdef USE_RETRO_ACHIEVEMENTS
+    Config::SetBaseOrCurrent(Config::RA_HARDCORE_ENABLED, m_settings.enable_hardcore);
```

Comment:
If I'm reading this correctly, because you're setting this in the base layer, the enabling/disabling of hardcode mode will linger around after the netplay session is over. Are you sure this is what you want? I would have expected this code to use the `layer` parameter instead.